src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -797,16 +821,8 @@ MulticopterPositionControl::control_manual(float dt)
 	/* vertical axis */
 	if (_control_mode.flag_control_altitude_enabled) {
 		/* check for pos. hold */
-		if (fabsf(req_vel_sp(2)) < _params.hold_z_dz) {
-			if (!_alt_hold_engaged) {
-				if (_params.hold_max_z < FLT_EPSILON || fabsf(_vel(2)) < _params.hold_max_z) {
```

Comment:
Me again ;) If we remove this, and have it directly go into altitude hold, what happens if you climb or descent at max speed and then do a "full stop" (centering throttle fast)? With the hold_max_z code in place it probably eases out better, and you can still disable that with the param.